web-common/src/features/explores/ExploreEditor.svelte

```diff
@@ -35,10 +36,12 @@
     bind:autoSave
     bind:editor
     onSave={(content) => {
+      const name = findResourceNameInYAML(content) ?? exploreName;
```

Comment:
We should fix `exploreName` at the top of the component tree, not just in this one place. It looks like `ExploreWorkspace.svelte` line 26 first defines `exploreName`. It should pull the name from the `meta` in the `GetResource` query, else maybe fallback to the filename.